wordpress-mobile/WordPress-iOS
# Add build and upload to iOS tools

This PR adds two lanes to manage the build and the upload to the right service of the Internal and Public versions of the app.

## To Test:
### Setup
1.  Update your `~/.wpios-env.default` file with all the new values from `env.example`;

### Internal build test:
HockeyApp allows us to delete uploaded binaries, so it's fine to build a "fake" app.
2 .modify the version in the config/Version.Internal.xcconfig file with a newest date;
3. commit your changes on the local branch;
4. `cd Scripts` and run `fastlane build_and_upload_internal`
5.  wait for the build and the upload to finish
6. open the Web HockeyApp and verify that the builds is there. Verify that the dSYM file is uploaded too. The script sets the "Available for Download" option off, so it won't be available to users.
7. delete the test binary from HockeyApp.

### TestFlight build test: 
ITC doesn't allow to delete a binary, so we'll test the connection and the fact that the binary is recognised.
8. Revert the changes you made at point 2;
9. `cd Scripts` and run `fastlane build_and_upload_itc`
10. wait for the build to complete and the upload to start;
11. enter your username when required;
12. the file will be uploaded, then an error will raised because the current version is already on TestFlight. This is expected and the error confirms that the binary was uploaded anyway.
